src/devices/QwiicButton/QwiicButton.cs

```diff
@@ -0,0 +1,96 @@
+//// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.I2c;
+using Iot.Device.Common;
+using Iot.Device.QwiicButton.RegisterMapping;
+
+namespace Iot.Device.QwiicButton
+{
+    /// <summary>
+    /// SparkFun Qwiic Button is an I2C based button with a built-in LED.
+    /// Supported hardware version: 1.0.0
+    /// </summary>
+    public sealed partial class QwiicButton : IDisposable
+    {
+        private const int DefaultAddress = 0x6F; // Default I2C address of the button
+        private I2cRegisterAccess<Register> _registerAccess;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QwiicButton"/> class.
+        /// </summary>
+        /// <param name="i2cBusId">I2C bus ID the button is connected to.</param>
+        /// <param name="i2cAddress">I2C bus address of the button (default=0x6F).</param>
+        public QwiicButton(int i2cBusId, byte i2cAddress = DefaultAddress)
+        {
+            I2cBusId = i2cBusId;
+            I2cAddress = i2cAddress;
+            var settings = new I2cConnectionSettings(i2cBusId, i2cAddress);
+            var device = I2cDevice.Create(settings);
+            _registerAccess = new I2cRegisterAccess<Register>(device);
+        }
+
+        /// <summary>
+        /// I2C bus ID the button is connected to.
+        /// </summary>
+        public int I2cBusId { get; set; }
+
+        /// <summary>
+        /// I2C bus address of the button.
+        /// </summary>
+        public byte I2cAddress { get; set; }
+
+        /// <summary>
+        /// Returns the 8-bit device ID of the button.
+        /// </summary>
+        public byte GetDeviceId()
+        {
+            return _registerAccess.ReadSingleRegister(Register.Id);
+        }
+
+        /// <summary>
+        /// Returns the firmware version of the button as a 16-bit integer.
+        /// The leftmost (high) byte is the major revision number, and the rightmost (low) byte is
+        /// the minor revision number.
+        /// </summary>
+        public ushort GetFirmwareVersionAsInteger()
+        {
+            ushort version = (ushort)(_registerAccess.ReadSingleRegister(Register.FirmwareMajor) << 8);
+            version |= _registerAccess.ReadSingleRegister(Register.FirmwareMinor);
+            return version;
+        }
+
+        /// <summary>
+        /// Returns the firmware version of the button as a [major revision].[minor revision] string.
+        /// </summary>
+        public string GetFirmwareVersionAsString()
+        {
+            var major = _registerAccess.ReadSingleRegister(Register.FirmwareMajor);
+            var minor = _registerAccess.ReadSingleRegister(Register.FirmwareMinor);
+            return major + "." + minor;
+        }
+
+        /// <summary>
+        /// Configures the button to attach to the I2C bus using the specified address.
+        /// </summary>
+        public void SetI2cAddress(byte address)
```

Comment:
Most of the devices become unusable after setting the address and need to be re-constructed, if that's the case here you should rename this to: `SetI2cAddressAndDispose` and dispose on usage

Reply:
Done